programs/locker/src/instructions/v2/create_vesting_escrow.rs

```diff
@@ -0,0 +1,114 @@
+use anchor_spl::token::spl_token;
+use anchor_spl::token_2022::spl_token_2022;
+use anchor_spl::token_interface::{Mint, TokenAccount, TokenInterface};
+
+use crate::util::{calculate_transfer_fee_included_amount, transfer_to_escrow_v2, validate_mint};
+use crate::TokenProgramFlag::{UseSplToken, UseToken2022};
+use crate::*;
+
+#[event_cpi]
+#[derive(Accounts)]
+pub struct CreateVestingEscrowV2<'info> {
+    #[account(mut)]
+    pub base: Signer<'info>,
+
+    #[account(
+        init,
+        seeds = [
+        b"escrow".as_ref(),
+        base.key().as_ref(),
+        ],
+        bump,
+        payer = sender,
+        space = 8 + VestingEscrow::INIT_SPACE
+    )]
+    pub escrow: AccountLoader<'info, VestingEscrow>,
+
+    pub token_mint: Box<InterfaceAccount<'info, Mint>>,
+
+    #[account(
+        mut,
+        associated_token::mint = sender_token.mint,
```

Comment:
pls use: associated_token::mint = token_mint

Reply:
updated, also validate the sender_token too

Reply:
as discussed, I revert the check for sender_token